tests/mir-opt/pre-codegen/loops.int_range.PreCodegen.after.mir

```diff
@@ -6,35 +6,33 @@ fn int_range(_1: usize, _2: usize) -> () {
     let mut _0: ();
     let mut _3: std::ops::Range<usize>;
     let mut _4: std::ops::Range<usize>;
-    let mut _5: &mut std::ops::Range<usize>;
-    let mut _11: std::option::Option<usize>;
-    let mut _14: isize;
-    let _16: ();
+    let mut _8: std::option::Option<usize>;
+    let mut _11: isize;
+    let _13: ();
+    let mut _14: &mut std::ops::Range<usize>;
     scope 1 {
         debug iter => _4;
-        let _15: usize;
+        let _12: usize;
         scope 2 {
-            debug i => _15;
+            debug i => _12;
         }
         scope 4 (inlined iter::range::<impl Iterator for std::ops::Range<usize>>::next) {
-            debug self => _5;
+            debug self => &_4;
             scope 5 (inlined <std::ops::Range<usize> as iter::range::RangeIteratorImpl>::spec_next) {
-                debug self => _5;
-                let mut _6: &usize;
-                let mut _7: &usize;
-                let mut _10: bool;
-                let _12: usize;
-                let mut _13: usize;
+                debug self => &_4;
+                let mut _7: bool;
+                let _9: usize;
+                let mut _10: usize;
                 scope 6 {
-                    debug old => _12;
+                    debug old => _9;
                     scope 7 {
                     }
                 }
                 scope 8 (inlined cmp::impls::<impl PartialOrd for usize>::lt) {
-                    debug self => _6;
-                    debug other => _7;
-                    let mut _8: usize;
-                    let mut _9: usize;
+                    debug self => &((*_14).0: usize);
+                    debug other => &((*_14).1: usize);
```

Comment:
Is it possible something got lost along the way in the debuginfo here?  I think there should have ended up as
```
                    debug self => _4.0: usize;
                    debug other => _4.1: usize;
```
since `_14` seems to never be initialized.

Reply:
This is very likely caused by DSE, which ignores debuginfo. I have a patch to correct that, waiting on your SimplifyLocals PR.